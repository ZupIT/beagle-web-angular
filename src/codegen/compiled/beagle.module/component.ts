/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Type } from '@angular/core'
import {
  kebabToCamelCase,
  removeExtraIndentation,
  replaceToUnderline,
} from '../../utils/formatting'
import { getComponentAnnotations } from '../../utils/metadata'
import { remoteViewSelector } from '../../../constants'


function createViewChildString(name: string, templateName: string, angularVersion: number): string {
  const componentName = replaceToUnderline(name)

  if (angularVersion >= 8) {
    return `'${componentName}': new ViewChild('${templateName}', { static: true })`
  } else {
    return `'${componentName}': new ViewChild('${templateName}')`
  }
}

function createQueries(components: Record<string, Type<any>>, angularVersion: number) {
  const componentNames = Object.keys(components)

  return componentNames.map((name) => {
    const selector = getComponentAnnotations(components[name]).selector
    const templateName = kebabToCamelCase(selector)
    return createViewChildString(name, templateName, angularVersion)
  })
}


export function createComponentString(
    components: Record<string, Type<any>>,
    angularVersion: number,
  ) {
  const queries = createQueries(components, angularVersion)

  // todo: legacy code. Remove the input "loadParams" with v2.0.
  const componentString = `
    @Component({
      selector: '${remoteViewSelector}',
      template,
      inputs: ['loadParams', 'route', 'networkOptions', 'controllerId'],
      queries: {
        ${queries.join(',\n        ')},
      },
    })
    export class BeagleRemoteView extends AbstractBeagleRemoteView {
      constructor(
        beagleProvider: BeagleProvider,
        ngZone: NgZone,
        changeDetector: ChangeDetectorRef,
      ) {
        // @ts-ignore
        super(beagleProvider, ngZone, changeDetector)
      }
    }
  `

  return removeExtraIndentation(componentString, 4)
}
