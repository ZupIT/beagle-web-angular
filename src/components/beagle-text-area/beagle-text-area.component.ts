/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Component, Input, OnInit } from '@angular/core'
import { logger } from '@zup-it/beagle-web'
import { TextAreaInterface } from '../schemas/text-area'
import { InputHandler } from '../schemas/input-handler'
import { BaseComponent } from '../../runtime/BaseComponent'

/**
 * @deprecated since version 1.8.0.
 * This component will be removed in a future version.
*/
@Component({
  selector: 'beagle-text-area',
  templateUrl: './beagle-text-area.component.html',
  styleUrls: ['./beagle-text-area.component.less'],
})
export class BeagleTextAreaComponent extends BaseComponent 
  implements TextAreaInterface, OnInit {
  @Input() value?: string
  @Input() label?: string
  @Input() name?: string
  @Input() onChange?: InputHandler
  @Input() onFocus?: InputHandler
  @Input() onBlur?: InputHandler
  @Input() disabled?: boolean
  @Input() readonly?: boolean

  ngOnInit() {
    this.label = this.label || ''
    this.disabled = this.disabled || false
    this.readonly = this.readonly || false
    logger.warn('This component is deprecated since version 1.8.0 and will be removed in a future version.')
  }

  public handleChange(value) {
    this.onChange && this.onChange({ value })
  }

  public handleBlur(event) {
    this.onBlur && this.onBlur({ value: event.target.value })
  }

  public handleFocus(event) {
    this.onFocus && this.onFocus({ value: event.target.value })
  }
}
