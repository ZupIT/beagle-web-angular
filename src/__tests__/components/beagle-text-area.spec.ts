/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { TestBed, async, ComponentFixture, fakeAsync } from '@angular/core/testing'
import { FormsModule } from '@angular/forms'
import { BeagleTextAreaComponent } from '../../components/beagle-text-area/beagle-text-area.component'
import { setAndCallHandler } from './mocks/test-mocks.spec'
import { buildBeagleTestModuleMetadata } from './test-module-builder'

let component: BeagleTextAreaComponent
let fixture: ComponentFixture<BeagleTextAreaComponent>

describe('BeagleTextAreaComponent', () => {

  beforeEach(async(() => {
    TestBed.configureTestingModule(
      buildBeagleTestModuleMetadata([BeagleTextAreaComponent], [FormsModule]))
      .compileComponents()
      
    fixture = TestBed.createComponent(BeagleTextAreaComponent)
    component = fixture.componentInstance
    component.value = 'Testing'
    component.label = 'Label Test'
    component.name = 'Label'
    component.onChange = jest.fn()
    component.onFocus = jest.fn()
    component.onBlur = jest.fn()

  }))

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('should call handlers', fakeAsync(() => {
    spyOn(component, 'handleChange').and.callThrough()
    spyOn(component, 'handleFocus').and.callThrough()
    spyOn(component, 'handleBlur').and.callThrough()

    setAndCallHandler('textarea', 'Testing', 'input', fixture)
    setAndCallHandler('textarea', 'Testing', 'focus', fixture)
    setAndCallHandler('textarea', 'Testing', 'blur', fixture)

    expect(component.onChange).toHaveBeenCalled()
    expect(component.onFocus).toHaveBeenCalled()
    expect(component.onBlur).toHaveBeenCalled()
  }))
})
