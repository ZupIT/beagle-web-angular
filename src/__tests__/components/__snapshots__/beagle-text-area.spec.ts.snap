// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeagleTextAreaComponent should match snapshot 1`] = `
BeagleTextAreaComponent {
  "disabled": true,
  "label": "Label Test",
  "name": "Label",
  "onBlur": [MockFunction],
  "onChange": [MockFunction],
  "onFocus": [MockFunction],
  "readonly": true,
  "value": "Testing",
}
`;
